File: leetcode/solutions/0374.GuessNumberHigherOrLower/guessNumber_test.go
package solution

import (
	"testing"
)

func Test_0374_GuessNumberHigherOrLower(t *testing.T) {
	t.Parallel()

	var val int

	val = 6
	answer.setAnswer(6)
	got := guessNumber(10)

	if got != val {
		t.Errorf("got %#v want %#v", got, val)
	}
}

type gamesAnswer struct {
	answer int
}

func (g *gamesAnswer) setAnswer(num int) {
	g.answer = num
}

var answer gamesAnswer

// Mock API "guess"
func guess(num int) int {
	switch {
	case answer.answer < num:
		return -1
	case answer.answer > num:
		return 1
	default:
		return 0
	}
}

File: leetcode/solutions/0374.GuessNumberHigherOrLower/guessNumber_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import guessNumber from "./guessNumber.ts";

let answer = -1;

export function guess(num: number): number {
  if (answer < num) {
    return -1;
  } else if (answer > num) {
    return 1;
  }

  return 0;
}

Deno.test("0374. Guess Number Higher or Lower", () => {
  answer = 6;
  assertEquals(guessNumber(10), 6);

  answer = 1;
  assertEquals(guessNumber(10), 1);

  const ans = 100;
  answer = ans;
  assertEquals(guessNumber(100), ans);
});
